core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -548,4 +550,63 @@ default HttpResponseDuplicator toDuplicator(EventExecutor executor, long maxResp
     }
+
+    /**
+     * Returns a new {@link SplitHttpResponse} which splits a stream of {@link HttpObject}s into
+     * {@link HttpHeaders} and {@link HttpData}.
+     * {@link SplitHttpResponse#headers()} will be
+     * completed before publishing the first {@link HttpData}.
+     * {@link SplitHttpResponse#trailers()} might not complete until the entire response body is consumed
+     * completely.
+     */
+    @CheckReturnValue
+    default SplitHttpResponse split() {
+        return split(defaultSubscriberExecutor());
+    }
+
+    /**
+     * Returns a new {@link SplitHttpResponse} which splits a stream of {@link HttpObject}s into
+     * {@link HttpHeaders} and {@link HttpData}.
+     * {@link SplitHttpResponse#headers()} will be completed before publishing the first {@link HttpData}.
+     * {@link SplitHttpResponse#trailers()} might not complete until the entire response body is consumed
+     * completely.
+     *
+     * <p>(Advanced users only) If you want to get pooled objects from {@link SplitHttpResponse#body()},
+     * you should call this method with {@link SubscriptionOption#WITH_POOLED_OBJECTS}.
+     * Note that if you specify {@link SubscriptionOption}s when subscribing to
+     * {@link SplitHttpResponse#body()}, {@link UnsupportedOperationException} will be raised.
+     */
+    @CheckReturnValue
+    default SplitHttpResponse split(SubscriptionOption... options) {
+        return new DefaultSplitHttpResponse(this, defaultSubscriberExecutor(), options);
+    }
+
+    /**
+     * Returns a new {@link SplitHttpResponse} which splits a stream of {@link HttpObject}s into
+     * {@link HttpHeaders} and {@link HttpData}.
+     * {@link SplitHttpResponse#headers()} will be completed before publishing the first {@link HttpData}.
+     * {@link SplitHttpResponse#trailers()} might not complete until the entire response body is consumed
+     * completely.
+     */
+    @CheckReturnValue
+    default SplitHttpResponse split(EventExecutor executor) {
+        return new DefaultSplitHttpResponse(this, executor);
+    }
+
+    /**
+     * Returns a new {@link SplitHttpResponse} which splits a stream of {@link HttpObject}s into
+     * {@link HttpHeaders} and {@link HttpData}.
+     * {@link SplitHttpResponse#headers()} will be completed before publishing the first {@link HttpData}.
+     * {@link SplitHttpResponse#trailers()} might not complete until the entire response body is consumed
+     * completely.
+     *
+     * <p>(Advanced users only) If you want to get pooled objects from {@link SplitHttpResponse#body()},
+     * you should call this method with {@link SubscriptionOption#WITH_POOLED_OBJECTS}.
+     * Note that if you specify {@link SubscriptionOption}s when subscribing to
+     * {@link SplitHttpResponse#body()}, {@link UnsupportedOperationException} will be raised.
+     */
+    @CheckReturnValue
+    default SplitHttpResponse split(EventExecutor executor, SubscriptionOption... options) {
```

Comment:
I'm not sure we need to accept `SubscriptionOption` here. A user will have to specify them when retrieving the response body via `SplitHttpResponse.body()`. Otherwise, a user will have to specify the same option twice.

We could subscribe to `HttpResponse` with both `SubscriptionOption`s enabled internally, and then emulate the desired behavior based on what's specified in `body.subscribe()`, e.g.

- Convert off-heap `HttpData` into heap `HttpData` if `WITH_POOLED_OBJECTS` option is off.
- Propagate `CancelledSubscriptionException` to `onError()` only when `NOTIFY_CANCELLATION` option is on.